[MRG] Link contributing guide

betatim opened the pull request:
This PR links the contributing guide more prominently.

i also changed the wording at the top of the contributing section a bit. This is taken from what we have in https://repo2docker.readthedocs.io/en/latest/contributing/contributing.html. For BinderHub we have a slightly different setup https://github.com/jupyterhub/binderhub#contributing

I think making it as easy as possible to find the contributing guide helps show people new to the project that there are several ways of contributing and that they aren't super difficult or require specialised knowledge. It is also useful for repeat contributors because they want a reminder of how things work (build the docs, run the tests, etc).

What do you think?

GeorgianaElena commented on README.rst:
```diff
@@ -14,14 +14,23 @@ The Littlest JupyterHub
+
+For support questions please search or post to `our forum <https://discourse.jupyter.org/c/jupyterhub/>`_.
+
+See the `contributing guide <https://the-littlest-jupyterhub.readthedocs.io/en/latest/contributing/index.html>`_`
```
```suggestion
See the `contributing guide <https://the-littlest-jupyterhub.readthedocs.io/en/latest/contributing/index.html>`_
```

GeorgianaElena commented on docs/contributing/index.rst:
```diff
@@ -2,16 +2,21 @@
+
+This section contains documentation for people who want to contribute.
+
+You can find the `source code on GitHub <>`_
```
```suggestion
You can find the `source code on GitHub <https://github.com/jupyterhub/the-littlest-jupyterhub/tree/master/tljh>`_
```

betatim commented:
I like that badge! Where can I get the "source" of it to then link it in the README?

manics commented:
Optional suggestion: Add a `CONTRIBUTING.md` that links to the published doc, so if someone opens an issue they'll see a link https://help.github.com/en/github/building-a-strong-community/setting-guidelines-for-repository-contributors#adding-a-contributing-file

GeorgianaElena commented:
@betatim, you can hit "edit" on my comment above to see the badge link.

betatim commented:
I added a `CONTRIBUTING.rst` (the readme is a rst already so stuck with that). I didn't know that this would add a link on the new issue page!

I added the "I want to contribute" badge to the `README.rst`. I hadn't realised that I can edit your comment and see the source :)

betatim commented:
I'll spend a bit of time looking at the failed integration test.

betatim commented:
My thoughts on the failing integration test are in #490.
